### ISSUE TITLE
Output should be able to be rotated.

### ISSUE THREAD

**Issue by TMaYaD** [Cloudef/wlc#86]
similar to -rotate option for xrandr.

**Comment by Cloudef** [comment]
Not sure if possible with drm (perhaps only with certain devices). Would be possible with OpenGL though at least.

**Comment by TMaYaD** [comment]
Complete noob to Wayland here... Does this help?
http://www.jlekstrand.net/jason/projects/wayland/transforms/

**Comment by TMaYaD** [comment]
Looks like it's more involved than that. https://ereleaseblog.wordpress.com/2015/05/15/fun-with-drm-and-randr/

Still no clue what hardware planes are. Need to figure it out next fortnight I guess. :)

**Comment by TMaYaD** [comment]
Also, is this related?
https://github.com/Cloudef/wlc/issues/64

**Comment by Cloudef** [comment]
No, buffer transformations are for clients. (They may request their buffer to be shown rotated for example, where compositor needs to handle input etc accordingly)

**Comment by harrisonrodgers** [comment]
Hi. Could the [transform](https://wayland.freedesktop.org/docs/html/apa.html#protocol-spec-wl_output-enum-transform) parameter of [wl_output](https://wayland.freedesktop.org/docs/html/apa.html#protocol-spec-wl_output) be what we want? 
> This describes the transform that a compositor will apply to a surface to compensate for the rotation or mirroring of an output device.

**Comment by Cloudef** [comment]
That's only for communicating the current rotation to client.

**Comment by WovenTales** [comment]
I haven't looked at the code of either this or Weston to know how compatible they are, but that does handle my rotated monitor correctly. If nothing else, that might point at one way to handle it.

**Comment by ohmysql** [comment]
This thread should give an idea how tricky this may prove to be:

https://bugzilla.gnome.org/show_bug.cgi?id=745079

**Comment by stevekuznetsov** [comment]
Looks like that BZ is marked fixed -- will WLC be able to support this feature in the future, then?

**Comment by AndreasBackx** [comment]
Any update on this issue?

**Comment by hnykda** [comment]
I am also interested - this is a bit blocker for me to currently use sway unfortunately :-(

**Comment by SirCmpwn** [comment]
As far as sway is concerned - we're working on switching from wlc to a new library, wlroots. It will take some time, but wlroots already supports output rotation.

**Comment by AndreasBackx** [comment]
Cool, I hope to switch from i3 to Sway as soon as I can then 👍

**Comment by hnykda** [comment]
Awesome! Thanks a lot. Do you have some time frame for this?

**Comment by SirCmpwn** [comment]
No.
